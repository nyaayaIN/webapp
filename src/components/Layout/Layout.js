import React from 'react';
import PropTypes from 'prop-types';
import withStyles from 'isomorphic-style-loader/lib/withStyles';

// external-global styles must be imported in your JS.
import normalizeCss from 'normalize.css';
import s from './Layout.css';
import Header from '../Header';
import Footer from '../Footer';
import FeaturedTopics from '../FeaturedTopics';

class Layout extends React.Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
  };

  render() {
    return (
      <div>
        <Header />
        {this.props.children}
        <FeaturedTopics />
        <Footer />
      </div>
    );
  }
}

export default withStyles(normalizeCss, s)(Layout);
