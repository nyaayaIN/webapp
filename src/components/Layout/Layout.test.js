/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Layout from './Layout';

global.fetch = require('jest-fetch-mock');

describe('Layout', () => {
  test('renders children correctly', () => {
    const wrapper = renderer
      .create(
        <App context={{ insertCss: () => {}, fetch: () => {} }}>
          <Layout>
            <div className="child" />
          </Layout>
        </App>,
      )
      .toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
