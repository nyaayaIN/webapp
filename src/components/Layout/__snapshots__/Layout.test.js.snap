// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout renders children correctly 1`] = `
<div>
  <div
    className="root"
  >
    <div
      className="container"
    >
      <div
        className="root"
        role="navigation"
      >
        <a
          className="link"
          href="/about-us"
          onClick={[Function]}
        >
          About Us
        </a>
        <a
          className="link"
          href="/contact"
          onClick={[Function]}
        >
          Contact Us
        </a>
        <a
          className="link"
          href="https://medium.com/nyaaya"
          rel="noopener noreferrer"
          target="_blank"
        >
          Blog
        </a>
      </div>
      <a
        className="brand"
        href="/"
        onClick={[Function]}
      >
        <img
          alt="Nyaaya"
          height="75"
          src="logo.png"
          srcSet="logo@2x.png 2x"
          width="95"
        />
      </a>
    </div>
    <div
      className="root"
    >
      <div
        className="container"
      >
        <ul
          className="categories"
        />
      </div>
    </div>
  </div>
  <div
    className="child"
  />
  <div
    className="root"
  >
    <div
      className="container"
    >
      <span
        className="text"
      >
        Nyaaya - India's Laws Explained
      </span>
    </div>
  </div>
</div>
`;
